<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1596</width>
    <height>1031</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Shannon-Fano coding</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="2" column="0" colspan="2">
     <widget class="QPushButton" name="getString">
      <property name="text">
       <string>Выбрать файл, содержащий строку</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Шеннон-Фано</string>
      </property>
      <widget class="QWidget" name="gridLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>711</width>
         <height>361</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="8" column="0">
         <widget class="QLabel" name="label_symcodes">
          <property name="text">
           <string>|Символ|Частота|Код</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QTextBrowser" name="text_alphabet"/>
        </item>
        <item row="10" column="0">
         <widget class="QPushButton" name="btn_ApplyEncode">
          <property name="text">
           <string>Рассчитать</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QTextBrowser" name="text_ViewEncodeString"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Шифр XOR</string>
      </property>
      <widget class="QWidget" name="gridLayoutWidget_4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>30</y>
         <width>711</width>
         <height>371</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Ключ</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="btn_XOR">
          <property name="text">
           <string>Расчитать</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="line_XOR"/>
        </item>
        <item row="0" column="1">
         <widget class="QTextBrowser" name="text_XOR"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Шифр Виженера</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <widget class="QWidget" name="gridWidget_3" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>701</width>
         <height>381</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="1">
         <widget class="QLineEdit" name="line_Key"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Ключ</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QTextBrowser" name="text_Vijener"/>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="btn_Vijener">
          <property name="text">
           <string>Рассчитать</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Шифр Цезаря</string>
      </property>
      <widget class="QWidget" name="gridLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>751</width>
         <height>371</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="2">
         <widget class="QLabel" name="label_Caesar">
          <property name="text">
           <string>Шифр Цезаря</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="btn_Caesar">
          <property name="text">
           <string>Рассчитать</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QSlider" name="caesarOffset">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>33</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QTextBrowser" name="caesarBrowser"/>
        </item>
       </layout>
      </widget>
      <widget class="QLineEdit" name="line_Caesar">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>410</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QPushButton" name="btn_Save">
      <property name="text">
       <string>Сохранить</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="2">
     <widget class="QPushButton" name="btn_Exit">
      <property name="text">
       <string>Закрыть</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1596</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>О программе</string>
    </property>
    <addaction name="action"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="text">
    <string>Справка</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
